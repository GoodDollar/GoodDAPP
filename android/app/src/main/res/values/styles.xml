<resources>

    <!-- Base application theme. -->
    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
        <!-- Customize your theme here. -->
        <item name="android:textColor">#000000</item>
        <item name="android:forceDarkAllowed">false</item>
    </style>

    <style name="SplashTheme" parent="Theme.AppCompat.DayNight.NoActionBar">
        <item name="android:windowBackground">@drawable/splash</item>
    </style>


</resources>
