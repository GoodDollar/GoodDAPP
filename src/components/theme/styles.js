import DefaultTheme from 'react-native-paper/src/styles/DefaultTheme'
import normalize from '../../lib/utils/normalizeText'

export const theme = {
  ...DefaultTheme,
  borders: {
    defaultBorderColor: '#c9c8c9',
  },
  colors: {
    ...DefaultTheme.colors,
    facebookBlue: '#1877F3',
    googleBlue: '#3367D6',
    blue: '#006EA0',
    darkBlue: '#0C263D',
    darkGray: '#42454A',
    gray: '#555',
    gray50Percent: '#CBCBCB',
    gray80Percent: '#A3A3A3',
    placeholder: '#CBCBCB',
    disabled: '#E3E3E2',
    green: '#00C3AE',
    lightGreen: '#00BBA5',
    lightGray: '#EEE',
    lighterGray: '#A3A3A3',
    orange: '#F8AF40',
    primary: '#00AFFF',
    purple: '#9F6A9D',
    red: '#FA6C77',
    googleRed: '#EA4335',
    error: '#FA6C77',
    text: '#222',
    white: '#ffffff',
  },
  fonts: {
    ...DefaultTheme.fonts,
    default: 'Roboto',
    slab: 'Roboto Slab',
  },
  paddings: {
    defaultMargin: 8,
    mainContainerPadding: 8,
  },
  sizes: {
    default: 8,
    defaultDouble: 16,
    defaultQuadruple: 32,
    defaultHalf: 4,
    borderRadius: 5,
    maxHeightForTabletAndDesktop: 844,
    minHeightForDialogMessage: 350,
    maxWidthForTabletAndDesktop: 475,
  },
  fontStyle: {
    color: '#555',
    fontSize: normalize(18),
    textAlign: 'center',
  },
  backdrop: {
    backgroundColor: 'rgba(255, 0, 0, 0.8)',
  },
  modals: {
    overlayBackgroundColor: 'transparent',
    activityIndicatorBackgroundColor: 'rgba(255, 255, 255, 0.8)',
    overlayHorizontalPadding: 20,
    overlayVerticalPadding: 65,
    backgroundColor: '#fff',
    jaggedEdgeSize: 16,
    contentPadding: 16,
    borderRadius: 5,
    borderLeftWidth: 10,
  },
  feedItems: {
    borderRadius: 8,
    itemBackgroundColor: '#fff',
    height: normalize(84),
  },
}
