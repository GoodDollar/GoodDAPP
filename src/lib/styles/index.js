import React from 'react'
import { useTheme, withTheme } from 'react-native-paper'
import { StyleSheet } from 'react-native'

import { isFunction } from 'lodash'

export const withOpacity = (color, opacity) => {
  let decimal = opacity

  if (!Number.isInteger(opacity) && opacity <= 1) {
    decimal = Math.round(255 * opacity)
  }

  return `${color}${decimal.toString(16)}`
}

/**
 * HOC that injects `theme` from `withTheme` and `styles` using theme values into the `Component`
 * @param {*} mapThemeToStyles receives props and retuns an object with calculated styles
 * @param {*} withStyleSheet wheather should or shouldn't be the result wrapped with `StyleSheet.create`
 */
export const withStyles = (mapThemeToStyles, withStyleSheet = true) => Component => {
  const getUpdatedStyles = props => {
    let styles = {}
    if (isFunction(mapThemeToStyles)) {
      const stylesObject = mapThemeToStyles(props)
      styles = withStyleSheet ? StyleSheet.create(stylesObject) : stylesObject
    }
    return styles
  }

  class WrappedComponent extends React.Component {
    constructor(props) {
      super(props)
      const styles = getUpdatedStyles(props)
      this.state = { styles }
    }

    static getDerivedStateFromProps(props, state) {
      const styles = getUpdatedStyles(props)
      return { styles }
    }

    _root

    /**
     * @internal
     * Required for when the component is wrapped inside a TouchableOpacity* element
     * https://facebook.github.io/react-native/docs/direct-manipulation
     */
    setNativeProps(...args) {
      return this._root && this._root.setNativeProps(...args)
    }

    render() {
      if (!isFunction(mapThemeToStyles)) {
        return <Component ref={c => (this._root = c)} {...this.props} />
      }

      return <Component {...this.props} styles={this.state.styles} />
    }
  }

  // Adding static methods to new wrapped component
  Object.getOwnPropertyNames(Component)
    .filter(prop => isFunction(Component[prop]))
    .forEach(staticMethod => (WrappedComponent[staticMethod] = Component[staticMethod]))
  WrappedComponent.navigationOptions = Component.navigationOptions

  return withTheme(WrappedComponent)
}

export const makeStyles = (mapThemeToStyles, withStyleSheet = true) => {
  const getUpdatedStyles = props => {
    let styles = {}

    if (isFunction(mapThemeToStyles)) {
      const stylesObject = mapThemeToStyles(props)

      styles = withStyleSheet ? StyleSheet.create(stylesObject) : stylesObject
    }

    return styles
  }

  const useStyles = props => {
    const theme = useTheme()

    return { theme, styles: getUpdatedStyles({ ...props, theme }) }
  }

  return useStyles
}
