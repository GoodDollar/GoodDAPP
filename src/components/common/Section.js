// @flow
import React, { Component } from 'react'
import { Text, View, StyleSheet } from 'react-native'
import normalize from 'react-native-elements/src/helpers/normalizeText'
import { fontStyle } from './styles'

const Row = (props: any) => (
  <View {...props} style={[styles.sectionRow, props.style]}>
    {props.children}
  </View>
)
const Title = (props: any) => (
  <Text {...props} style={[styles.title, props.style]}>
    {props.children}
  </Text>
)
const SectionText = (props: any) => <Text {...props} style={[styles.text, props.style]} />
const Separator = () => <hr style={{ width: '100%' }} />

export default class Section extends Component<any> {
  static Row = Row
  static Title = Title
  static Text = SectionText
  static Separator = Separator

  render() {
    return <View style={[styles.section, this.props.style]}>{this.props.children}</View>
  }
}

const styles = StyleSheet.create({
  section: {
    backgroundColor: '#eeeeef',
    borderRadius: normalize(5),
    padding: normalize(10),
    paddingTop: normalize(15),
    paddingBottom: normalize(15),
    marginBottom: normalize(15)
  },
  sectionRow: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  title: {
    ...fontStyle
  },
  text: {
    ...fontStyle,
    fontSize: normalize(14)
  }
})
