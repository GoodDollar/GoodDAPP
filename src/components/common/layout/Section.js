// @flow
import React from 'react'
import { View } from 'react-native'
import { withStyles } from '../../../lib/styles'
import SectionRow from './SectionRow'
import SectionStack from './SectionStack'
import SectionTitle from './SectionTitle'
import SectionText from './SectionText'
import Separator from './Separator'
import { getFlexStylesFromProps } from './SectionUtils'

const SectionComponent = props => {
  const { styles, children, style, ...rest } = props
  return (
    <View style={[styles.section, getFlexStylesFromProps(props), style]} props={rest}>
      {children}
    </View>
  )
}

const mapPropsToStyles = ({ theme }) => ({
  section: {
    backgroundColor: theme.colors.surface,
    borderRadius: theme.sizes.borderRadius,
    paddingHorizontal: 12,
    paddingVertical: theme.sizes.defaultDouble,
  },
})

const Section = withStyles(mapPropsToStyles)(SectionComponent)

Section.Title = SectionTitle
Section.Row = SectionRow
Section.Text = SectionText
Section.Stack = SectionStack
Section.Separator = Separator

export default Section
