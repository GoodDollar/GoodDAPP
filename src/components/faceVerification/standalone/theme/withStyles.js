// utils
import { getDesignRelativeHeight, getDesignRelativeWidth, isLargeDevice } from '../../../../lib/utils/sizes'
import normalize from '../../../../lib/utils/normalizeText'
import { withStyles } from '../../../../lib/styles'
import { isBrowser } from '../../../../lib/utils/platform'

const getStylesFromProps = ({ theme }) => ({
  topContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: theme.colors.surface,
    borderRadius: theme.sizes.borderRadius,
    display: 'flex',
    flexDirection: 'column',
    flexShrink: 0,
    paddingBottom: getDesignRelativeHeight(theme.sizes.defaultDouble),
    paddingLeft: getDesignRelativeWidth(theme.sizes.default),
    paddingRight: getDesignRelativeWidth(theme.sizes.default),
    paddingTop: getDesignRelativeHeight(theme.sizes.defaultDouble),
  },

  mainContent: {
    flexGrow: 1,
    justifyContent: 'center',
    width: '100%',
  },

  descriptionContainer: {
    paddingHorizontal: getDesignRelativeHeight(theme.sizes.defaultHalf),
    paddingVertical: getDesignRelativeHeight(isBrowser ? theme.sizes.defaultDouble : 14),
    alignItems: 'center',
  },
  descriptionContainerB: {
    paddingVertical: getDesignRelativeHeight(isBrowser ? 12 : 10),
  },
  descriptionWrapper: {
    flexDirection: 'column',
    alignItems: 'center',
  },

  text: {
    textAlign: 'center',
    fontSize: normalize(isLargeDevice ? 22 : 20),
    lineHeight: isLargeDevice ? 36 : 34,
  },

  infoRow: {
    justifyContent: 'flex-start',
    alignItems: 'center',
  },

  action: {
    width: '100%',
  },
  actionsSpace: {
    marginBottom: getDesignRelativeHeight(16),
  },
})

export default withStyles(getStylesFromProps)
