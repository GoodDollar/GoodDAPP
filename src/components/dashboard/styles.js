import { StyleSheet } from 'react-native'
import normalize from '../../lib/utils/normalizeText'
import { theme } from '../theme/styles'
import { getScreenHeight } from '../../lib/utils/orientation'

const isMobileHeight = getScreenHeight() < 680

export const receiveStyles = StyleSheet.create({
  wrapper: {
    justifyContent: 'flex-start',
    width: '100%',
    padding: 8,
  },
  section: {
    flex: 1,
  },
  sectionRow: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  qrCode: {
    marginTop: isMobileHeight ? 0 : '2rem',
    padding: '1rem',
    borderColor: '#555',
    borderWidth: 1,
    borderRadius: '4px',
  },
  addressSection: {
    marginBottom: isMobileHeight ? 0 : '1rem',
  },
  address: {
    margin: '0.5rem',
  },
  secondaryText: {
    margin: isMobileHeight ? '0.2rem' : '1rem',
    color: '#555555',
    fontSize: normalize(14),
    textTransform: 'uppercase',
  },
  lowerSecondaryText: {
    margin: '1rem',
    color: '#555555',
    fontSize: normalize(16),
  },
  headline: {
    ...theme.fontStyle,
    textTransform: 'uppercase',
    marginBottom: '1rem',
    fontSize: normalize(24),
  },
  buttonGroup: {
    width: '100%',
    flexDirection: 'row',
    marginTop: '1rem',
  },
  amountLabel: {
    ...theme.fontStyle,
    fontSize: normalize(32),
  },
  amountSymbol: {
    fontSize: normalize(12),
  },
  shareQRButton: {
    width: '100%',
  },
  doneButton: {
    marginTop: '1rem',
  },
  fullWidth: {
    marginHorizontal: 10,
  },
  amountSuffix: {
    flexGrow: 1,
    height: 40,
    fontSize: normalize(10),
    justifyContent: 'center',
    lineHeight: 40,
    paddingTop: 10,
  },
  buttonStyle: {
    marginTop: '1em',
  },
})
