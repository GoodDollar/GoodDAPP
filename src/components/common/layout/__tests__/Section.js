import React from 'react'
import renderer from 'react-test-renderer'
import { Provider as PaperProvider } from 'react-native-paper'
import Section from '../Section'
import { theme } from '../../../theme/styles'

// Note: test renderer must be required after react-native.

describe('Section', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <PaperProvider theme={theme}>
        <Section>
          <Section.Title>GoodDollar is a good economy, each day you can collect your part in the economy</Section.Title>
          <Section.Row>
            <Section.Text>{`TODAY'S DAILY INCOME `}</Section.Text>
          </Section.Row>
        </Section>
      </PaperProvider>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot with properties', () => {
    const component = renderer.create(
      <PaperProvider theme={theme}>
        <Section>
          <Section.Stack grow>
            <Section.Text>Second text</Section.Text>
          </Section.Stack>
          <Section.Row alignItems="flex-end">
            <Section.Text>{`TODAY'S DAILY INCOME `}</Section.Text>
          </Section.Row>
        </Section>
      </PaperProvider>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
