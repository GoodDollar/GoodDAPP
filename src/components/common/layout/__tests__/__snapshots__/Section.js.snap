// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n r-backgroundColor-14lw9ot r-borderRadius-1f0042m r-paddingHorizontal-1e081e0 r-paddingVertical-1yzf0co"
  >
    <div
      className="css-text-901oao r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-hiw28u r-fontWeight-1it3c9n r-lineHeight-qvk6io r-marginVertical-c8eef1 r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-tsynxw"
      dir="auto"
    >
      GoodDollar is a good economy, each day you can collect your part in the economy
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
    >
      <div
        className="css-text-901oao r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-ubezar r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        TODAY'S DAILY INCOME 
      </div>
    </div>
  </div>
</div>
`;

exports[`Section matches snapshot with properties 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n r-backgroundColor-14lw9ot r-borderRadius-1f0042m r-paddingHorizontal-1e081e0 r-paddingVertical-1yzf0co"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1oszu61 r-flexDirection-eqz5dr r-justifyContent-1wtj0ep"
      style={
        Object {
          "WebkitBoxFlex": 1,
          "WebkitFlexGrow": 1,
          "flexGrow": 1,
          "msFlexPositive": 1,
        }
      }
    >
      <div
        className="css-text-901oao r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-ubezar r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        Second text
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
      style={
        Object {
          "WebkitAlignItems": "flex-end",
          "WebkitBoxAlign": "end",
          "alignItems": "flex-end",
          "msFlexAlign": "end",
        }
      }
    >
      <div
        className="css-text-901oao r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-ubezar r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        TODAY'S DAILY INCOME 
      </div>
    </div>
  </div>
</div>
`;
